server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1233,6 +1237,20 @@ public QueryDataSource query(String deviceId, String measurementId, QueryContext
       if (filePathsManager != null) {
         filePathsManager.addUsedFilesForQuery(context.getQueryId(), dataSource);
       }
+
+      // exclude repetitive tsFile to calculate the number of it
+      if (config.isEnablePerformanceTracing()) {
+        if (seqFile == null) {
+          seqFile = new HashSet<>(seqResources);
```

Comment:
If I query multiple times, when do you clear this?

Reply:
Fixed.